project-codeflare/codeflare-sdk
# refactor: codeflare sdk unit tests

sort unit tests into individual files and made minor enhancements

# Issue link


Closes: [RHOAIENG-13847](https://issues.redhat.com/browse/RHOAIENG-13847)
# What changes have been made

* Refactored unit tests into separate files from `unit_test.py`
* Updated `get_ray_obj` and `get_aw_obj` related helper functions (No more full RayCluster/AppWrapper yamls in a test file!!!)
* Made minor enhancements to pre-existing unit tests
* Merged all `ClusterConfiguration` related tests into one test that tests all parameters (instead of multiple smaller tests that test 1 or 2)
* removed demo-tests.py and func-tests.py (I think these were initial test files which we have never used)


# Verification steps



* Create a shell with `poetry shell`
* Install the CodeFlare SDK -> `poetry install --with test,docs`
* Run all of the tests using `coverage run -m --source=src pytest src/codeflare_sdk --ignore=src/codeflare_sdk/common/utils/unit_test_support.py`
* Check coverage with `coverage report` or `coverage report -m | tail -1 | tail -c 4`
## Checks
- Testing Strategy
